hs_core/models.py

```diff
@@ -55,6 +55,46 @@
 from .languages_iso import languages as iso_languages
 
 
+def clean_abstract(original_string):
+    """Clean abstract for XML inclusion.
+
+    This function takes an original string and removes any illegal XML characters
+    from it. It uses regular expressions to identify and remove the illegal characters.
+
+    Args:
+        original_string (str): The original string to be cleaned.
+
+    Returns:
+        str: The cleaned string with illegal XML characters removed.
+
+    Raises:
+        ValidationError: If there is an error cleaning the abstract.
+
+    """
+    # https://stackoverflow.com/a/64570125
+    try:
```

Comment:
Yes I lifted this without modification :)

I was going to use `xml.sax.saxutils escape` but it doesn't catch all of the invalids. This seemed more thorough

Reply:
realizing I might need to add replacements for quotes and ampersands...

> replace("&", "&amp;")
replace("<", "&lt;")
replace(">", "&gt;")
replace('"', "&quot;")
replace("'", "&apos;")

Reply:
GH renders those characters automatically
![image](https://github.com/hydroshare/hydroshare/assets/17934193/eba76713-737a-47ab-b573-d430b779098b)

Reply:
This is probably too late into the process... but couldn't we also encode any offending characters when writing the xml?

Reply:
Yes! We could remove the form validation/cleaning and just sanitize at the time of XML generation. I don't know if there could be unintended consequences for a slight mismatch between our HS metadata and the XML that gets deposited with Crossref...I can't think of any issues

Reply:
The only time that could be an issue is on bag ingestion, which is rare.  I think since it is already done, we should just go with this.  I don't see any issues with this approach and your implementation.

Reply:
The examples you have in this thread... aren't those url encodings?

Reply:
good point re bag ingestion, thanks

Reply:
yes you're right

Reply:
I verified that the XML generated for all the chars mentioned in this thread _is acceptable for deposit with Crossref_